app/policies/pet_policy.rb

```diff
@@ -4,4 +4,16 @@ def resolve
       scope.all
     end
   end
+
+  def update?
+    record.user == user
+  end
+
+  def show?
+    true
+  end
+
+  def edit?
+    update?
+  end
```

Comment:
It can be removed as edit inherits from ApplicationPolicy. And in there it checks for create